pkg/extensions/sync/sync_test.go

```diff
@@ -298,6 +286,195 @@ func TestSyncOnDemand(t *testing.T) {
 	})
 }
 
+func TestSyncOnDemandContentFiltering(t *testing.T) {
```

Comment:
Cover all permutations of onDemand (set/unset), pollInterval(set/unset) and content (set/unset)